include/lbann/callbacks/confusion_matrix.hpp

```diff
@@ -87,18 +86,18 @@ class confusion_matrix : public data_type_callback<TensorDataType> {
    *  This is a CPU matrix. If the prediction layer keeps data on GPU,
    *  then this will be a matrix copy rather than a matrix view.
    */
-  std::unique_ptr<El::AbstractDistMatrix<TensorDataType>> m_predictions_v;
+  std::unique_ptr<El::AbstractDistMatrix<DataType>> m_predictions_v;
```

Comment:
So this is where I ran into issues before, which led me to the state that the code is in right now.  This is a class that seems like it needs to be partially templated.